contracts/src/v0.8/ccip/pools/SiloedLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,286 @@
+      s_siloedChainConfigs[adds[i].remoteChainSelector].rebalancer = adds[i].rebalancer;
+
+      // Emit a siloing event to indicate that the chain is now siloed, but that the amount of liquidity
+      // is zero until liquidity is provided.
+      emit ChainSiloed(adds[i].remoteChainSelector);
+    }
+  }
+
+  /// @notice Gets the rebalancer able to provide liquidity for a remote chain selector
+  /// @param remoteChainSelector The CCIP specific selector for the remote chain being interacted with.
+  /// @return The current liquidity manager, contract owner if the chain's funds are not siloed.
+  function getRebalancerByChain(
+    uint64 remoteChainSelector
+  ) public view returns (address) {
+    ChainSiloConfig memory remoteConfig = s_siloedChainConfigs[remoteChainSelector];
+    if (remoteConfig.isSiloed) return remoteConfig.rebalancer;
+    else return s_unsiloedChainRebalancer;
+  }
+
+  /// @notice Sets the Rebalancer address for a given remoteChainSelector.
+  /// @dev Only callable by the owner.
+  /// @param remoteChainSelector the remote chain to set.
+  /// @param newRebalancer the address allowed to add liquidity for the given siloed chain.
+  function setSiloedChainRebalancer(uint64 remoteChainSelector, address newRebalancer) external onlyOwner {
+    ChainSiloConfig memory remoteConfig = s_siloedChainConfigs[remoteChainSelector];
+
+    if (!remoteConfig.isSiloed) revert ChainNotSiloed(remoteChainSelector);
+
+    address oldRebalancer = remoteConfig.rebalancer;
+
+    s_siloedChainConfigs[remoteChainSelector].rebalancer = newRebalancer;
+
+    emit SiloedChainRebalancerSet(remoteChainSelector, newRebalancer, oldRebalancer);
+  }
+
+  /// @notice Sets the Rebalancer address for unsiloed chains.
+  /// @dev Only callable by the owner.
+  /// @param newRebalancer the address allowed to add liquidity for the given siloed chain.
+  function setUnsiloedChainRebalancer(
+    address newRebalancer
+  ) external onlyOwner {
+    address oldRebalancer = s_unsiloedChainRebalancer;
+
+    s_unsiloedChainRebalancer = newRebalancer;
+
+    emit UnsiloedChainRebalancerSet(newRebalancer, oldRebalancer);
+  }
+
+  /// @notice Adds liquidity to the pool. The tokens should be approved first.
+  /// @param remoteChainSelector the remote chain to set. If the chain is not siloed, the liquidity will be shared among all
+  /// non-siloed chains.
+  /// @param amount The amount of liquidity to provide.
+  /// @dev Only the rebalancer for the chain can add liquidity
+  function provideSiloedLiquidity(uint64 remoteChainSelector, uint256 amount) external {
+    _provideLiquidity(remoteChainSelector, amount);
+  }
+
+  /// @notice Adds liquidity to the pool for unsiloed chains. Function is used to support legacy liquidity operations
+  /// by using a function selector available to previous L/R pools.
+  /// @dev Since the remoteChainSelector 0 should never be applied to a real chain, it is used to designate unsiloed chains.
```

Comment:
> remoteChainSelector 0 should never be applied to a real chain

We can even make sure that 0 is not treated as valid in the `updateSiloDesignationForChainSelectors` function to ensure it's correct

Reply:
Stale Comment. Removing.